prymer/primer3/primer3.py

```diff
@@ -710,21 +711,38 @@ def _build_failures(
             by_fail_count[Primer3FailureReason.LONG_DINUC] = num_dinuc_failures
         return [Primer3Failure(reason, count) for reason, count in by_fail_count.most_common()]
 
-    def _pad_target_region(self, target: Span, max_amplicon_length: int) -> Span:
+    def _create_design_region(
+        self,
+        target: Span,
+        max_amplicon_length: int,
+        min_primer_length: int,
+    ) -> Span:
         """
-        If the target region is smaller than the max amplicon length, pad to fit.
+        Construct a design region surrounding the target region.
+
+        The target region is padded on both sides by the maximum amplicon length, minus the length
+        of the target region itself.
+
+        If the target region cannot be padded by at least the minimum primer length on both sides,
+        a `ValueError` is raised.
+
+        Raises:
+            ValueError: If the target region is too large to be padded.
 
-        When the max amplicon length is odd, the left side of the target region will be padded with
-        one more base than the right side.
         """
+        padding: int = max_amplicon_length - target.length
         contig_length: int = self._dict[target.refname].length
-        padding_right: int = max(0, int((max_amplicon_length - target.length) / 2))
-        padding_left: int = max(0, max_amplicon_length - target.length - padding_right)
 
-        region: Span = replace(
+        if padding < min_primer_length:
```

Comment:
I think you want to move this check below creation of the `design_region` and check to ensure that the design_region start/end are at least min_primer_length </> the target's.  Otherwise it won't catch the case where the padding gets limited by the contig start/end.